external_parser/unit_tests/test_vw_external_parser.cc

```diff
@@ -87,6 +87,69 @@ BOOST_AUTO_TEST_CASE(ccb_simple) {
   VW::finish(*vw);
 }
 
+BOOST_AUTO_TEST_CASE(slates_simple) {
+  std::string input_files = get_test_files_location();
+
+  auto buffer = read_file(input_files + "/valid_joined_logs/slates_simple.log");
+
+  auto vw = VW::initialize("--ccb_explore_adf --slates --binary_parser --quiet", nullptr,
+                           false, nullptr, nullptr);
+
+  v_array<example *> examples;
+  examples.push_back(&VW::get_unused_example(vw));
+
+  set_buffer_as_vw_input(buffer, vw);
+
+  bool read_payload = false;
+
+  while (vw->example_parser->reader(vw, examples) > 0) {
+    read_payload = true;
+
+    BOOST_CHECK_EQUAL(examples.size(), 9);
+    BOOST_CHECK_EQUAL(examples[0]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[0]->indices[0], 'G');
+    BOOST_CHECK_EQUAL(examples[1]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[1]->indices[0], 'T');
+    BOOST_CHECK_EQUAL(examples[2]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[2]->indices[0], 'T');
+    BOOST_CHECK_EQUAL(examples[3]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[3]->indices[0], 'T');
+    BOOST_CHECK_EQUAL(examples[4]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[4]->indices[0], 'T');
+    BOOST_CHECK_EQUAL(examples[5]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[5]->indices[0], 'T');
+    BOOST_CHECK_EQUAL(examples[6]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[6]->indices[0], 'S');
+    BOOST_CHECK_EQUAL(examples[7]->indices.size(), 1);
+    BOOST_CHECK_EQUAL(examples[7]->indices[0], 'S');
+    BOOST_CHECK_EQUAL(examples[8]->indices.size(), 0); // newline example
+
+    multi_ex multi_exs;
+    for (auto *ex : examples) {
+      multi_exs.push_back(ex);
+    }
+
+    vw->learn(multi_exs);
+
+    examples[0]->pred.decision_scores = {
```

Comment:
nice! maybe add a comment saying that the slates clear example expects the pred to be set